dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -70,66 +72,162 @@ class Identity[A] extends Transformer[A, A] {
+      System.arraycopy(padArr, padOffset, data, offset, length)
+      offset += length
+    }
+  }
+
+  private def paddingValue(data: Array[T], padValue: T, start: Int, end: Int): Unit = {
+    ev.getType() match {
+      case DoubleType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Double]], start, end, ev.toType[Double](padValue))
+      case FloatType =>
+        util.Arrays.fill(data.asInstanceOf[Array[Float]], start, end, ev.toType[Float](padValue))
+      case _ => throw new UnsupportedOperationException(
+        "SampleToBatchPadding: Only Float/Double supported")
+    }
+  }
+
+  private def getSize(padLength: Int, batchLength: Int, tensorSize: Array[Int]): Int = {
+    require(tensorSize.length >= 2)
+    tensorSize(0) = 1
+    tensorSize(1) = padLength
+    val tensorLength = tensorSize.product
+    tensorSize(0) = batchLength
+    tensorLength
+  }
+
   override def apply(prev: Iterator[Sample[T]]): Iterator[MiniBatch[T]] = {
     new Iterator[MiniBatch[T]] {
-      private var featureTensor: Tensor[T] = Tensor[T]()
-      private var labelTensor: Tensor[T] = Tensor[T]()
+      private val featureTensor: Tensor[T] = Tensor[T]()
+      private val labelTensor: Tensor[T] = Tensor[T]()
       private var featureData: Array[T] = null
       private var labelData: Array[T] = null
+      private var sampleData: Array[Sample[T]] = null
+
       private val batchSize = Utils.getBatchSize(totalBatch)
       private var featureSize: Array[Int] = null
       private var labelSize: Array[Int] = null
       private var oneFeatureLength: Int = 0
       private var oneLabelLength: Int = 0
+      private var padFeatureLength: Int = 0
+      private var padLabelLength: Int = 0
+      private val padFeature: Boolean = if (padFeatureValue.isEmpty) false else true
+      private val padLabel: Boolean = if (padLabelValue.isEmpty) false else true
       override def hasNext: Boolean = prev.hasNext
 
       override def next(): MiniBatch[T] = {
         if (prev.hasNext) {
           var i = 0
+          var (maxfeatureElement, labelIndex) = (0, 0)
+          var (maxlabelElement, featureIndex) = (0, 0)
+          var padLength: Int = 0
+          if (sampleData == null) sampleData = new Array[Sample[T]](batchSize)
           while (i < batchSize && prev.hasNext) {
             val sample = prev.next()
-            if(!sample.feature().isContiguous() || !sample.label().isContiguous()) {
-              throw new IllegalArgumentException(
-                "Only support contiguous tensor, pls use tensor.contiguous() before batching")
+            require(sample.feature().isContiguous() && sample.label().isContiguous(),
+              "SampleToBatchPadding: Only support contiguous tensor, pls use " +
+                "tensor.contiguous() before batching")
+            if (sampleData(i) == null) sampleData(i) = Sample()
+            if ((maxfeatureElement != 0) && (!padFeature)) {
+              require(sample.feature().nElement() == maxfeatureElement,
```

Comment:
The check is not correct. Ideally we need to check if all samples in the batch have the same size (instead of same element#).